src/lightning_app/utilities/packaging/cloud_compute.py

```diff
@@ -1,5 +1,42 @@
 from dataclasses import asdict, dataclass
-from typing import List, Optional, Union
+from typing import Dict, List, Optional, Union
+from uuid import uuid4
+
+from lightning_app.core.constants import MULTI_WORKS_INTO_SINGLE_POD
+
+__CLOUD_COMPUTE_IDENTIFIER__ = "__cloud_compute__"
+
+
+@dataclass
+class _CloudComputeStore:
+    id: str
+    component_names: List[str]
+    frozen: bool = False
+
+    def add_component_name(self, new_component_name: str) -> None:
+        found_index = None
+        for index, component_name in enumerate(self.component_names):
+            if new_component_name.endswith(component_name.replace("root.", "")):
```

Comment:
can you comment here what is this doing? It is be hard for a new reader to figure this out

Reply:
We are popping out the previous comment name.